website/blog/2025-03-14-20-must-try-mongodb-queries-operations-ferretdb.md

```diff
@@ -0,0 +1,463 @@
+    "established": 1780
+  },
+  {
+    "_id": 2,
+    "name": "Harper & Brothers",
+    "location": "New York, United States",
+    "established": 1817
+  },
+  {
+    "_id": 3,
+    "name": "Lackington, Hughes, Harding, Mavor & Jones",
+    "location": "London, United Kingdom",
+    "established": 1790
+  }
+]
+```
+
+You can join the data from both collections (`books` and `publishers`) using `$lookup`.
+
+```js
+db.books.aggregate([
+  {
+    $lookup: {
+      from: 'publishers',
+      localField: 'publication.publisher.name',
+      foreignField: 'name',
+      as: 'publisher_details'
+    }
+  }
+])
+```
+
+This should fetch the books and their corresponding publishers' details in a single query.
+
+## Authentication
+
+### 8. Create an authenticated user
+
+Authentication is important for securing your database and ensuring only authorized users can access it, especially for production-based environments.
+FerretDB relies entirely on PostgreSQL authentication mechanisms.
+When you create a user in FerretDB, you can manage it using the same commands you would on MongoDB.
+
+You can create a user in FerretDB as you would on MongoDB by just running the `createUser` command.
+
+For example, the following command creates a `newuser` with password `newpassword` with all the credentials stored on PostgreSQL.
+
+```js
+db.createUser({
+  user: 'newuser',
+  pwd: 'newpassword',
+  roles: []
+})
+```
+
+You can learn more about authentication in FerretDB [here](https://docs.ferretdb.io/security/authentication/).
+
+### 9. Delete user
+
+What if you no longer need a particular user account?
+For example, you can remove the newly created `newuser` account on FerretDB simply by running the command:
+
+```js
+db.dropUser('newuser')
+```
+
+## Indexes
+
+### 10. Create indexes and drop indexes
+
+Speed is always a hot topic for databases.
+Indexes help speed up queries, and that's crucial for better database performance.
+
+Indexes create optimized data structures (sort of like a table of contents) that store references to document locations.
+That way, queries can jump straight to the data via the document locations instead of scanning entire collections.
+
+On FerretDB, you can create an index on a book's title and price:
+
+```js
+db.books.createIndex({ title: 1, 'price.value': -1 })
+```
+
+If you no longer need an index, you can drop it to free up space and resources.
+
+For instance, to drop the index on the `title` field:
+
+```js
+db.books.dropIndex({ title: 1 })
+```
+
+Learn more about indexes in FerretDB [here](https://docs.ferretdb.io/usage/indexes/).
+
+### 11. TTL indexes
+
+A Time-To-Live (TTL) index auto-cleans expired data after a set time.
+That means no manual deletions and no bloated storage.
+
+If a library tracks book reservations for instance, a TTL index can remove records after a year, so outdated holds don't clutter queries or waste space.
+
+Let's say we want books to be removed automatically one year after publication.
```

Comment:
You had such a great example in the first sentence, and then you ruined it by removing the book instead

Reply:
What's wrong with removing the book? :thinking:

Reply:
I believe the concept of expiring books from publication date is not a good example?

Reply:
Yep. Since books technically don't expire but it's now updated as books reservations, which can actually expire